mailbox/event/event-rabbitmq/src/test/java/org/apache/james/mailbox/events/RabbitMQEventBusTest.java

```diff
@@ -0,0 +1,243 @@
+            Sender sender = RabbitFlux.createSender(senderOption);
+
+            sender.declareQueue(QueueSpecification.queue(MAILBOX_WORK_QUEUE_NAME)
+                .durable(DURABLE)
+                .exclusive(!EXCLUSIVE)
+                .autoDelete(!AUTO_DELETE)
+                .arguments(NO_ARGUMENTS))
+                .block();
+            sender.bind(BindingSpecification.binding()
+                .exchange(MAILBOX_EVENT_EXCHANGE_NAME)
+                .queue(MAILBOX_WORK_QUEUE_NAME)
+                .routingKey(EMPTY_ROUTING_KEY))
+                .block();
+        }
+
+        @Test
+        void dispatchShouldPublishSerializedEventToRabbitMQ() {
+            eventBus.dispatch(EVENT, NO_KEYS).block();
+
+            assertThat(dequeueEvent()).isEqualTo(EVENT);
+        }
+
+        @Test
+        void dispatchShouldPublishSerializedEventToRabbitMQWhenNotBlocking() {
+            eventBus.dispatch(EVENT, NO_KEYS);
+
+            assertThat(dequeueEvent()).isEqualTo(EVENT);
+        }
+
+        private Event dequeueEvent() {
+            RabbitMQConnectionFactory connectionFactory = rabbitMQExtension.getConnectionFactory();
+            Receiver receiver = RabbitFlux.createReceiver(new ReceiverOptions().connectionMono(Mono.just(connectionFactory.create())));
+
+            byte[] eventInBytes = receiver.consumeAutoAck(MAILBOX_WORK_QUEUE_NAME)
+                .blockFirst()
+                .getBody();
+
+            return eventSerializer.fromJson(new String(eventInBytes, StandardCharsets.UTF_8))
+                .get();
+        }
+    }
+
+    @Nested
+    class ConcurrentTest {
+
+        private final Duration FIVE_SECONDS = Duration.ofSeconds(5);
+        private final ConditionFactory AWAIT_CONDITION = await().timeout(new com.jayway.awaitility.Duration(5, TimeUnit.SECONDS));
+
+        private static final int THREAD_COUNT = 10;
+        private static final int OPERATION_COUNT = 30;
+        private static final int TOTAL_DISPATCH_OPERATIONS = THREAD_COUNT * OPERATION_COUNT;
+
+        private final List<EventBusTestFixture.MailboxListenerCountingSuccessfulExecution> ALL_LISTENERS =
+            ImmutableList.of(COUNTING_LISTENER_1, COUNTING_LISTENER_2, COUNTING_LISTENER_3);
+
+        private int totalEventsReceived() {
+            return ALL_LISTENERS.stream()
+                .map(listener -> listener.numberOfEventCalls())
+                .reduce(Integer::sum)
+                .get();
+        }
+
+        private void runConcurrent(ConcurrentTestRunner.ConcurrentOperation operation) throws Exception {
+            ConcurrentTestRunner.builder()
+                .operation(operation)
+                .threadCount(THREAD_COUNT)
+                .operationCount(OPERATION_COUNT)
+                .runSuccessfullyWithin(FIVE_SECONDS);
+        }
+
+        @BeforeEach
+        void setUp() {
+            ALL_LISTENERS.forEach(listener -> listener.clear());
```

Comment:
I would rather re-create all the listeners rather than cleaning them (IE remove that .clear() method)